digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -235,10 +235,46 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
 
   def setRoadAddressFloating(ids: Set[Long]): Unit = {
     withDynTransaction {
-      ids.foreach(id => RoadAddressDAO.changeRoadAddressFloating(float = true, id))
+      // TODO: add geometry if it is somehow available
+      ids.foreach(id => RoadAddressDAO.changeRoadAddressFloating(float = true, id, None))
     }
   }
 
+  /*
+    Kalpa-API methods
+  */
+
+  def getRoadAddressesLinkByMunicipality(municipality: Int): Seq[RoadAddressLink] = {
+    val linkIdsAmmounts = 100
+    var roadLinks = roadLinkService.getViiteRoadLinksFromVVHByMunicipality(municipality)
+    val linkIds = roadLinks.map(_.linkId).toSet
+    val linkIdGroups = linkIds.grouped(linkIdsAmmounts)
+
+    val addresses = linkIdGroups.map(links =>
+      withDynTransaction {
+      RoadAddressDAO.fetchByLinkId(links)
+    }).flatten.toList.groupBy(_.linkId)
```

Comment:
We should get those links in one transaction if possible and with one query if possible. So not for each group but get them all from the database at once and then use map to combine the database loaded items with the link id groups.